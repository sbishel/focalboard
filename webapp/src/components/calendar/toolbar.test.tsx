// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React from 'react'
import {render, screen} from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import '@testing-library/jest-dom'

import {wrapIntl} from '../../testUtils'

import CustomToolbar from './toolbar'

describe('components/calendar/toolbar', () => {
    const mockNavigate = jest.fn()
    const mockView = jest.fn()

    beforeEach(() => {
        jest.clearAllMocks()
    })

    const localizer = {
        messages: {
            month: 'month',
            week: 'week',
            today: 'today',
        },
    }

    test('return toolbar', () => {
        const {container} = render(
            wrapIntl(
                <CustomToolbar
                    date={new Date()}
                    view='month'
                    views={['week', 'month']}
                    label='September 2021'
                    localizer={localizer}
                    onNavigate={mockNavigate}
                    onView={mockView}
                />,
            ),
        )
        expect(container).toMatchSnapshot()
    })

    test('return toolbar, no views', () => {
        const {container} = render(
            wrapIntl(
                <CustomToolbar
                    date={new Date()}
                    view='month'
                    views={[]}
                    label='September 2021'
                    localizer={localizer}
                    onNavigate={mockNavigate}
                    onView={mockView}
                />,
            ),
        )
        expect(container).toMatchSnapshot()
    })

    test('toolbar test navigate', () => {
        const {container} = render(
            wrapIntl(
                <CustomToolbar
                    date={new Date()}
                    view='month'
                    views={['week', 'month']}
                    label='September 2021'
                    localizer={localizer}
                    onNavigate={mockNavigate}
                    onView={mockView}
                />,
            ),
        )
        expect(container).toMatchSnapshot()
        const buttonPrev = screen.getByRole('button', {name: '<'})
        userEvent.click(buttonPrev)
        const buttonNext = screen.getByRole('button', {name: '>'})
        userEvent.click(buttonNext)
        const buttonToday = screen.getByRole('button', {name: localizer.messages.today})
        userEvent.click(buttonToday)

        expect(mockView).not.toBeCalled()
        expect(mockNavigate).toBeCalledTimes(3)
    })

    test('toolbar test view', () => {
        const {container} = render(
            wrapIntl(
                <CustomToolbar
                    date={new Date()}
                    view='month'
                    views={['week', 'month']}
                    label='September 2021'
                    localizer={localizer}
                    onNavigate={mockNavigate}
                    onView={mockView}
                />,
            ),
        )
        expect(container).toMatchSnapshot()
        const buttonMonth = screen.getByRole('button', {name: localizer.messages.month})
        userEvent.click(buttonMonth)

        const buttonWeek = screen.getByRole('button', {name: localizer.messages.week})
        userEvent.click(buttonWeek)

        expect(mockNavigate).not.toBeCalled()
        expect(mockView).toBeCalledTimes(2)
    })
})
