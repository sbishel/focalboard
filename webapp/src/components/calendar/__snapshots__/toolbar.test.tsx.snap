// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/calendar/toolbar return toolbar 1`] = `
<div>
  <div
    class="rbc-toolbar"
  >
    <span
      class="rbc-toolbar-label"
    >
      September 2021
    </span>
    <span
      class="rbc-btn-group"
    >
      <button
        class="rbc-btn-view"
        type="button"
      >
        week
      </button>
      <button
        class="rbc-btn-view rbc-active"
        type="button"
      >
        month
      </button>
    </span>
    <span
      class="rbc-btn-group"
    >
      <button
        type="button"
      >
        &lt;
      </button>
      <button
        type="button"
      >
        today
      </button>
      <button
        type="button"
      >
        &gt;
      </button>
    </span>
  </div>
</div>
`;

exports[`components/calendar/toolbar return toolbar, no views 1`] = `
<div>
  <div
    class="rbc-toolbar"
  >
    <span
      class="rbc-toolbar-label"
    >
      September 2021
    </span>
    <span
      class="rbc-btn-group"
    />
    <span
      class="rbc-btn-group"
    >
      <button
        type="button"
      >
        &lt;
      </button>
      <button
        type="button"
      >
        today
      </button>
      <button
        type="button"
      >
        &gt;
      </button>
    </span>
  </div>
</div>
`;

exports[`components/calendar/toolbar toolbar test navigate 1`] = `
<div>
  <div
    class="rbc-toolbar"
  >
    <span
      class="rbc-toolbar-label"
    >
      September 2021
    </span>
    <span
      class="rbc-btn-group"
    >
      <button
        class="rbc-btn-view"
        type="button"
      >
        week
      </button>
      <button
        class="rbc-btn-view rbc-active"
        type="button"
      >
        month
      </button>
    </span>
    <span
      class="rbc-btn-group"
    >
      <button
        type="button"
      >
        &lt;
      </button>
      <button
        type="button"
      >
        today
      </button>
      <button
        type="button"
      >
        &gt;
      </button>
    </span>
  </div>
</div>
`;

exports[`components/calendar/toolbar toolbar test view 1`] = `
<div>
  <div
    class="rbc-toolbar"
  >
    <span
      class="rbc-toolbar-label"
    >
      September 2021
    </span>
    <span
      class="rbc-btn-group"
    >
      <button
        class="rbc-btn-view"
        type="button"
      >
        week
      </button>
      <button
        class="rbc-btn-view rbc-active"
        type="button"
      >
        month
      </button>
    </span>
    <span
      class="rbc-btn-group"
    >
      <button
        type="button"
      >
        &lt;
      </button>
      <button
        type="button"
      >
        today
      </button>
      <button
        type="button"
      >
        &gt;
      </button>
    </span>
  </div>
</div>
`;
